st2api/st2api/controllers/exp/inquiries.py

```diff
@@ -0,0 +1,273 @@
+# Licensed to the StackStorm, Inc ('StackStorm') under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+from oslo_config import cfg
+
+import copy
+
+from six.moves import http_client
+from st2common.models.db.auth import UserDB
+from st2api.controllers.resource import ResourceController
+from st2api.controllers.v1.executionviews import SUPPORTED_FILTERS
+from st2common import log as logging
+from st2common.util import action_db as action_utils
+from st2common.util import schema as util_schema
+from st2common.util import system_info
+from st2common.services import executions
+from st2common.constants import action as action_constants
+from st2common.rbac.types import PermissionType
+from st2common.rbac import utils as rbac_utils
+from st2common.router import abort
+from st2common.router import Response
+from st2common.models.api.inquiry import InquiryAPI, InquiryResponseAPI
+from st2common.persistence.execution import ActionExecution
+from st2common.services import action as action_service
+from st2common.util.action_db import (get_action_by_ref, get_runnertype_by_name)
+
+from st2actions.container.base import get_runner_container
+
+__all__ = [
+    'InquiriesController'
+]
+
+LOG = logging.getLogger(__name__)
+INQUIRY_RUNNER = 'inquirer'
+
+
+class InquiriesController(ResourceController):
+    """API controller for Inquiries
+    """
+
+    supported_filters = copy.deepcopy(SUPPORTED_FILTERS)
+
+    # No data model currently exists for Inquiries, so we're "borrowing" ActionExecutions
+    # for the DB layer
+    model = InquiryAPI
+    access = ActionExecution
+
+    def get_all(self, requester_user=None, limit=None, **raw_filters):
+        """Retrieve multiple Inquiries
+
+            Handles requests:
+                GET /inquiries/
+        """
+
+        raw_inquiries = super(InquiriesController, self)._get_all(
+            limit=limit,
+            raw_filters={
+                'status': action_constants.LIVEACTION_STATUS_PENDING,
+                'runner': INQUIRY_RUNNER
+            }
+        )
+
+        # Transform to InquiryResponseAPI model
+        inquiries = [InquiryResponseAPI.from_inquiry_api(InquiryAPI.from_dict(raw_inquiry))
+                     for raw_inquiry in json.loads(raw_inquiries.body)]
```

Comment:
This is the first time I've seen that we have to handle json load for a record individually. Can you elaborate?

Reply:
I am trying to get the link to the actual thread that I was in with @Kami and @lakshmi-kannan further up in this PR but Github isn't letting me see back that far right now for some reason.

Basically there are two API models for Inquiries - one that actually satisfies the API definition, and another that has a few extra fields that I use to make decisions (like how [`_inquiry_sanity_check` uses `runner` and `status`](https://github.com/StackStorm/st2/pull/3653/files#diff-361872b73d29fa927dc3081ae5f6c082R190)). Since _get_all has already converted to JSON, and the `InquiriesController.model` is `InquiryAPI`, I needed to convert them all to `InquiryResponseAPI` instances. This seems the most succinct way to do that.

That's a lot of detail that the comment `Transform to InquiryResponseAPI model` doesn't quite cover, so I'll add something more there.

Reply:
I addressed this in what I think is a better way in be287e75c. In short, I got rid of the extra `from_dict` function. It was basically `from_model`, just without the call to `cls._from_model`. So I just went with a sort of hybrid function `from_model` function that allows this step to be skipped, and that way I can convert these `InquiryAPI` JSON objects directly into `InquiryResponseAPI` instances directly, simplifying this comprehension quite a bit.

Reply:
This is a potential performance problem to have to iterate thru the record and do a json loads...